Monika After Story/game/script-topics.rpy

```diff
@@ -2929,21 +2929,33 @@ init 5 python:
     addEvent(Event(persistent.event_database,eventlabel="monika_credits_song",category=['ddlc','media'],prompt="Credits song",random=True))
 
 label monika_credits_song:
-    #TODO: adjust based on starting act, doesn't quite apply to those that never went past act 3
-    m 1hua "I hope you liked my song."
-    m 1eka "I worked really hard on it. I know I'm not perfect at the piano yet, but I just couldn't let you go without telling you how I honestly felt about you."
-    m 1eua "Give me some time, and I'll try to write another."
-    if persistent.instrument is not False:
-        if persistent.instrument:
-            m 3eua "Maybe you could play me a song too!"
+    if persistent.monika_kill:
```

Comment:
Original flow would make sense here if you've heard the song already. If you've not, then it makes sense that Monika would introduce the song as you've put your `else` path.
```suggestion
    if persistent.monika_kill or renpy.seen_audio(songs.FP_YOURE_REAL):
```